c10/core/ScalarType.h

```diff
@@ -424,6 +454,10 @@ static inline bool isBitsType(ScalarType t) {
       t == ScalarType::Bits16;
 }
 
+static inline bool isBarebonesUnsignedType(ScalarType t) {
```

Comment:
Should UInt8 be in here as well?
I guess not given how you use it for the type promotion checks. But this might be a confusing API for our c++ devs?

Reply:
Bare bones here is defined to be "we have only minimal kernel support in traditional C++ eager mode". Since uint8 is grandfathered from Lua days to have lots of kernels, it shouldn't be included.

Reply:
What is the plan to document that to make sure user expectations for these are appropriate?

Reply:
The most logical place to document that these are bare bones is in torch.Tensor where we list supported dtypes. Note that the f8 dtypes are not documented right now, so we could also keep it under the radar undocumented until enough stuff is working. I am also half expecting to end up having a select few eager kernels sprout arbitrary unsigned support, especially gather-like operations I think is what people are most likely to want.

Reply:
One could argue that fp8 dtypes are not very well documented and shouldn't be the example here. Also there is a strong alignment there that we will not implement any arithmetic op for these dtypes (no scale means no compute with the value).
These dtypes could contain the full thing so we most likely want to have some details about that?

Reply:
I'll post a proposed doc PR and we can discuss it there

Reply:
https://github.com/pytorch/pytorch/pull/116804